docker-build-env/Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM ubuntu:focal
+
+# Install MoCOCrW dependencies (except OpenSSL)
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y --no-install-recommends install \
+        ca-certificates \
+        clang \
+        cmake \
+        g++ \
+        googletest \
+        libboost-all-dev \
+        make \
+        ninja-build \
+        pkg-config \
+        wget \
+        && rm -rf /var/lib/apt/lists/*
+
+# Verify that the OpenSSL dev package has not been installed inadvertently
+RUN ! dpkg -s libssl-dev
+
+# Build and install specific OpenSSL version
+RUN wget -O openssl.tar.gz https://github.com/openssl/openssl/archive/OpenSSL_1_1_1-stable.tar.gz \
```

Comment:
Should we really use an unversioned tarball here? I know it reduces our maintenance effort when security fixes are merged, but I guess our caching layer wouldn't rebuild the container anyway unless we touch the Dockerfile, so we wouldn't get the newer releases without manual intervention anyway?

Use of a specific release would also allow us to verify a checksum here. @garw, any opinions?

Reply:
I've based this on the [status quo](https://github.com/blenk92/MoCOCrW-build-env/blob/master/Dockerfile), assuming that it's been the preferred way.

I've since discussed it with Max, and it seems unclear whether there was any good reason for doing this in the first place. Is there actually a desire to pin a specific version here and not just use the one provided by Ubuntu 20.04? If so, I agree that pinning it to a repository version and not just to 1.1.1 would make sense. If not, we could just remove this altogether.

Reply:
https://github.com/blenk92/MoCOCrW-build-env/commit/6af34872fc6faa62005fd0093756546385a2c744 switched from the Ubuntu package to self-installing. I don't see a documented reason for this, but I'm assuming it's because Ubuntu didn't ship 1.1.1 (but 1.1.0) back then.

Maybe we should re-visit this now that focal has 1.1.1.

Reply:
Just to be clear, beware that simply installing libssl-dev yields essentially the same behavior as the current state of the PR: The actual version that ends up being used depends on when the image is built (not recalled from cache). Whether this is a bad thing depends on how much we trust that security fixes will not bring any surprises for MoCOCrW.

In similar scenarios, I've resorted to using Debian with snapshot.debian.org in order to really pin down everything. I'm not aware of an equivalent service for Ubuntu, and more importantly, as you already said, it adds maintenance effort.

Reply:
I think this was added because some features (I think some SHA3 support) was only available in a newer version than the one that came with ubuntu. In my opinion we can continue in two ways
1) We just take Ubuntu's OpenSSL version.
2) We use a fixed version and bump it every once in a while.

My gut feeling is that 1) is good enough for this purpose here and less maintainence effort. We use the internal-CI anyways additionally to check against the version that we internally use.

Reply:
Makes sense to me. I've replaced it all with `libssl-dev` in the dependencies list.